var/spack/repos/builtin/packages/warpx/package.py

```diff
@@ -423,43 +209,35 @@ def cmake_args(self):
             args.append("-DWarpX_openpmd_internal=OFF")
 
         if "+python" in spec:
-            if spec.satisfies("@24.08:"):
-                args.append("-DWarpX_pyamrex_internal=OFF")
-                args.append("-DWarpX_pybind11_internal=OFF")
-                args.append(self.define_from_variant("WarpX_PYTHON_IPO", "python_ipo"))
+            pip_args = PythonPipBuilder.std_args(self)
+            pip_args_install_index = pip_args.index("install")
+            pip_args_before_install = pip_args[:pip_args_install_index]
+            pip_args_after_install = pip_args[pip_args_install_index + 1 :] + [
+                f"--prefix={self.prefix}"
+            ]
+
+            args.append("-DWarpX_pyamrex_internal=OFF")
+            args.append("-DWarpX_pybind11_internal=OFF")
+            args.append(self.define_from_variant("WarpX_PYTHON_IPO", "python_ipo"))
+            # Additional parameters to pass to `pip`
+            args.append('-DPY_PIP_OPTIONS="' + " ".join(pip_args_before_install) + '"'),
+            # Additional parameters to pass to `pip install`
+            args.append('-DPY_PIP_INSTALL_OPTIONS="' + " ".join(pip_args_after_install) + '"'),
```

Comment:
This causes:
```py
python3 -m pip "-vvv\ --no-input\ --no-cache-dir\ --disable-pip-version-check" wheel --no-build-isolation --no-deps --wheel-dir=amrex-whl ...
no such option: -\
```
This is caused by `-vvv\`. Those `\` are added by the cmake build to escape the spaces.

This means that you must remove the quotes `"` around the string that you add:
Those prevent these backslashes from getting consumed by escape expansion, and thus they reach `pip`, which you do not want.

In GitLab CI, I also saw in https://gitlab.spack.io/spack/spack/-/jobs/12901412:
```py
Error: Failed to install py-amrex due to TypeError: write() argument must be str, not list
```
To make this code more readable, consistently use f-strings instead of string concatenation.